File: frontend/projects/api/src/lib/assessment.ts
import type { AssessmentMapping } from './assessment-mapping';
import type { AssessmentSelection } from './assessment-selection';
import type { QuestionPool } from './question-pool';

export interface Assessment {
  id: string;
  title: string;
  purpose: string;
  selection: AssessmentSelection;
  thresholdPercent: number;
  maximumAttempts: number;
  timeLimitMinutes: number | null;
  feedbackRelease: string;
  mappings: AssessmentMapping;
  pools: QuestionPool[];
}

File: backend/src/RepoFluent.Application/Assessment.cs
namespace RepoFluent.Application;

public sealed record Assessment(
    string Id,
    string Title,
    string Purpose,
    AssessmentSelection Selection,
    int ThresholdPercent,
    int MaximumAttempts,
    int? TimeLimitMinutes,
    string FeedbackRelease,
    AssessmentMapping Mappings,
    IReadOnlyList<QuestionPool> Pools);
